.github/workflows/hashicorp.yml

```diff
@@ -70,10 +74,16 @@ jobs:
     # Custom runners range in size from 4 core to 64 core and sizes `small` through `xl` are supported
     runs-on: [custom, linux, large]
     steps:
-      - uses: actions/checkout@v3
+      - if: inputs.git-ref == ''
```

Comment:
Actually, for the HashiCorp release process we should probably double-down on the already required `product-version` input, rather than allowing two inputs that could mismatch. I'm not sure if there is a case where there shouldn't be a Git tag that matches what product version we want to publish.

Reply:
I agree with you that we should probably enforce the `product-version` input, the problem is that since `product-version` is required, we will always be overriding the checkout by checking out the `product-version` instead of the default SHA that triggered the action. It's probably fine for most use cases but it will be a change in functionality. The `git-ref` is really only there to preserve backwards compatiblity.

Reply:
Maybe the solution is to create a new action/file instead of forcing this functionality into this workflow?

Reply:
Can you walk me through what you mean by this?

> It's probably fine for most use cases but it will be a change in functionality. The `git-ref` is really only there to preserve backwards compatiblity.

Goreleaser should already require a Git tag for a non-snapshot release on the Git commit being checked out. Using the product version (the same Git tag) for Git checkout should be functionally equivalent to pulling the latest commit from the default Git branch. The main difference I'm seeing is that using product version here _enables_ releases from non-default Git branches, since the checkout will occur on the Git tag matching whatever Git branch the release might be coming from. I'm not sure its taking away or breaking anything.

The code comment here:

https://github.com/hashicorp/ghaction-terraform-provider-release/blob/ef2f0c92ac9b5b15fd802308798f6e3e00840e53/.github/workflows/hashicorp.yml#L75

Relates to how the workflow was kicked off. It's technically possible for this workflow to be called from a workflow that was triggered by the push of a commit rather than a tag, but callers would already need to set product version to an appropriate Git tag they create in that case.

Reply:
I guess this will only be a concern if someone is trying to do a release with a release tag that is not the same as the product-version but I can't think of a practical reason for that so we can stick to enforcing the product-version parameter.